### ISSUE TITLE
Explicitly select tests to be run in kerberized product tests environments

### ISSUE THREAD

**Issue by losipiuk** [prestosql/presto#4333]
Currently we are running running tests in kerberized environments according to these rules:

https://github.com/prestosql/presto/blob/1b4275037b6f834dcb230d9ec9e6f028564216b5/presto-product-tests/bin/product-tests-suite-2.sh#L10-L28

The selected set of groups translates to large number of tests, not all relevant to the kerberization aspect of the environment.

Instead of using a numbercommon test-groups here, we should revisit which tests actually make sense to be run and tag those with `hdfs_impersonation` and/or `hdfs_no_impersonation` test group.
Then we can run tests only against those groups. Or at minimum drop `storage_formats` which is very time consuming.

Split-out from: #4176

**Comment by findepi** [comment]
> Instead of using a numbercommon test-groups here, we should revisit which tests actually make sense to be run and tag those with `hdfs_impersonation` and/or `hdfs_no_impersonation` test group.

I agree except i would introduce a  new test group for them, not to mix these 
tests with the tests actually testing that impersonation does happen.
Maybe `storage_authorization` or `authorization_smoke`?
